indico/modules/rb/client/js/modules/calendar/CalendarListView.jsx

```diff
@@ -101,23 +124,86 @@ class CalendarListView extends React.Component {
           {moment(day, 'YYYY-MM-DD').format('dddd, LL')}
         </Header>
         <Card.Group itemsPerRow={4} stackable styleName="day-cards-container">
-          {bookings.map(this.renderBooking)}
+          {bookings.map(booking => this.renderBooking(booking, day, linkData))}
         </Card.Group>
         <Divider hidden />
       </div>
     );
   };
 
-  renderBooking = booking => {
+  renderLink = (booking, day, linkData, linkBookingOccurrence) => {
+    if (linkData) {
+      const {linkingConfirmOpen} = this.state;
+      const {reservation} = booking;
+      const boundaries = [moment(linkData.startDt), moment(linkData.endDt)];
+      const datesMatch =
+        moment(booking.startDt).isBetween(...boundaries, undefined, '[]') &&
+        moment(booking.endDt).isBetween(...boundaries, undefined, '[]');
+      if (!!booking.link || !datesMatch) {
```

Comment:
When searching I see a huge amount of bookings that aren't valid linking candidates.

- Should we really show those, or add an extra filter to only see those occurrences where linking is actually possible? As a user I'd rather just see the ones I can actually link...
- Does it make sense to allow linking to someone else's booking?
- Shouldn't we filter by room by default if the event has a room set? If you want another one you can always clear the filter in the RB UI...

Reply:
> Should we really show those, or add an extra filter to only see those occurrences where linking is actually possible? As a user I'd rather just see the ones I can actually link...

I guess that you mean here an extra date filter so that only occurrences during the dates of the object to be linked? If so, for now, we would just show occurrences without adding the UI for the filter.

> Does it make sense to allow linking to someone else's booking?

I think we should only allow linking occurrences that are by/for the logged in user, unless the booking has an occurrence linked to the event already.

> Shouldn't we filter by room by default if the event has a room set? If you want another one you can always clear the filter in the RB UI

This would be an nice UX improvement. It could be done not only if events have a room but also if session blocks or contributions have one. Would you be ok with adding this on a second iteration?